presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionProcess.java

```diff
@@ -0,0 +1,266 @@
+                planFragmentCodec,
+                taskUpdateRequestCodec,
+                serverInfoCodec,
+                taskManagerConfig.getInfoRefreshMaxWait());
+        this.taskManagerConfig = requireNonNull(taskManagerConfig, "taskManagerConfig is null");
+        this.systemConfig = requireNonNull(systemConfig, "systemConfig is null");
+        this.nodeConfig = requireNonNull(nodeConfig, "nodeConfig is null");
+        this.connectorConfig = requireNonNull(connectorConfig, "connectorConfig is null");
+        this.nativeExecutionTask = new NativeExecutionTask(session, planFragment, sources, this.workerClient, tableWriteInfo, executor, updateScheduledExecutor, taskManagerConfig);
+        this.errorRetryScheduledExecutor = requireNonNull(errorRetryScheduledExecutor, "errorRetryScheduledExecutor is null");
+        this.errorTracker = nativeExecutionRequestErrorTracker("NativeExecution", uri, maxErrorDuration, errorRetryScheduledExecutor, "getting native process status");
+    }
+
+    /**
+     * Starts the external native execution process. The method will be blocked by connecting to the native process's /v1/info endpoint with backoff retries until timeout.
+     */
+    public void start()
+            throws ExecutionException, InterruptedException, IOException
+    {
+        String executablePath = getProcessWorkingPath(SystemSessionProperties.getNativeExecutionExecutablePath(session));
+        String configPath = Paths.get(executablePath, String.valueOf(port)).toAbsolutePath().toString();
+
+        populateConfigurationFiles(configPath);
+        ProcessBuilder processBuilder = new ProcessBuilder(executablePath, "--v", "1", "--etc_dir", configPath);
+        processBuilder.redirectOutput(ProcessBuilder.Redirect.INHERIT);
+        processBuilder.redirectError(ProcessBuilder.Redirect.INHERIT);
+        try {
+            log.info("Launching %s \nConfig path: %s\n", executablePath, configPath);
+            process = processBuilder.start();
+        }
+        catch (IOException e) {
+            throw new PrestoException(NATIVE_EXECUTION_PROCESS_LAUNCH_ERROR, format("Cannot start %s", processBuilder.command()), e);
+        }
+
+        getServerInfoWithRetry().get();
+    }
+
+    public NativeExecutionTask getTask()
+    {
+        return nativeExecutionTask;
+    }
+
+    @VisibleForTesting
+    public SettableFuture<ServerInfo> getServerInfoWithRetry()
+    {
+        SettableFuture<ServerInfo> future = SettableFuture.create();
+        doGetServerInfo(future);
+        return future;
+    }
+
+    public void close()
+    {
+        if (process.isAlive()) {
+            process.destroy();
+        }
+    }
+
+    private URI getBaseUriWithPort(URI baseUri, int port)
+    {
+        return uriBuilderFrom(baseUri)
+                .port(port)
+                .build();
+    }
+
+    private int getAvailableTcpPort()
+            throws IOException
+    {
+        int port = NATIVE_PROCESS_DEFAULT_PORT;
```

Comment:
This is assigned but always overwritten, is this assignment required ?

Reply:
You're right, it's not required, just removed.